flink-ml-lib/src/main/java/org/apache/flink/ml/feature/binarizer/Binarizer.java

```diff
@@ -0,0 +1,156 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.binarizer;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.ml.api.Transformer;
+import org.apache.flink.ml.common.datastream.TableUtils;
+import org.apache.flink.ml.linalg.DenseVector;
+import org.apache.flink.ml.linalg.SparseVector;
+import org.apache.flink.ml.linalg.typeinfo.DenseVectorTypeInfo;
+import org.apache.flink.ml.linalg.typeinfo.SparseVectorTypeInfo;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.util.ParamUtils;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A Transformer that binarizes the columns of continuous features by the given thresholds. The
+ * continuous features may be DenseVector, SparseVector, or Numerical Value.
+ */
+public class Binarizer implements Transformer<Binarizer>, BinarizerParams<Binarizer> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public Binarizer() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        String[] inputCols = getInputCols();
+        TypeInformation<?>[] outputTypes = new TypeInformation[inputCols.length];
+
+        for (int i = 0; i < inputCols.length; ++i) {
+            int idx = inputTypeInfo.getFieldIndex(inputCols[i]);
+            if (inputTypeInfo.getFieldTypes()[idx] instanceof SparseVectorTypeInfo) {
+                outputTypes[i] = SparseVectorTypeInfo.INSTANCE;
+            } else if (inputTypeInfo.getFieldTypes()[idx] instanceof DenseVectorTypeInfo) {
+                outputTypes[i] = DenseVectorTypeInfo.INSTANCE;
+            } else {
```

Comment:
nit: another possibility is that `inputTypeInfo.getFieldTypes()[idx] instanceof VectorTypeInfo`.